---
output:
    pdf_document:
        latex_engine: xelatex
        keep_tex: false
        includes:
          in_header: preamble_report.tex
        toc: true
fontsize: 11pt
geometry: margin=1in
bibliography: refs_report.bib
csl: mbio.csl
link-citations: true
title: "Investigating the microbial community of the human gut in colorectal cancer"
author: "Nicole Bowers, Brittany Hicks, Christina Kang-Yun, Katelyn Polemi, Kelly Sovacool"
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, 
                      fig.path = here::here("figures/"))
```
```{r load}
library(broom)
library(here)
library(knitr)
library(tidyverse)
```

\newpage
\todo{Guidelines: https://drive.google.com/file/d/1YNqYiMJYiVWXiSR9yWUbQTPAoUpVa9Id/view}
# Abstract

Colorectal cancer is the third leading cause of cancer related deaths globally [@rawla_epidemiology_2019]. 
Colorectal cancer (CRC) initiates in the large intestine emerging from glandular epithelial cells. 
Due to a selective advantage, obtained from a series of genetic or epigenetic mutations, these unregulated cells grow and can potentially develop into colorectal adenocarcinoma [@ewing_molecular_2014]. 
Many studies have used the cancer microbiome to investigate how these mutations occur, however, they have almost exclusively focused on bacteria. 
Human viruses have been implicated in the development of many cancers such as HPV, HTLV-1 and HIV [@liao_viruses_2006]. 
Through mutagenic and manipulative abilities, viruses cause unregulated growth by disrupting the normal function of cells. 
The association of colorectal cancer and the gut virome remains unknown, therefore, in this study we propose to investigate the differences in the bacterial and viral community composition and their effect on CRC development. 
We performed operational taxonomic unit (OTU) clustering of 16S sequences for bacterial taxonomy and operational viromic unit (OVU) clustering of metaviromes for viral taxonomy to investigate the differences in human colorectal bacterial and viral community composition. 
Using machine learning classification of OTUs and OVUs from healthy or cancerous samples, we aim to compare performance of the random forest model vs logistic regression. 
Overall, the goal of the current study is to classify whether samples are healthy or cancerous based on bacterial and viral taxa. These results provide… 

# Introduction

Colorectal cancer is a leading cause of cancer-related death in the United States and worldwide. Its risk and severity have been linked to colonic bacterial community composition
The leading cancer-related death in the United States is colorectal cancer [@hannigan_diagnostic_2018]. 
Due to screening techniques and improvements in treatment the rate of colorectal cancer has fallen. 
The primary screening technique is colonoscopy. A new technique gaining popularity is Exact Sciences Cologuard, a combination test which looks for both the genetic material found in colon cancer and some more advanced colon polyps and it detects hemoglobin in the stool using fecal immunochemical testing (Cologuard). 
While colon cancer screening techniques are advancing, we aim to understand the environment of colorectal cancer.

Gut dysbiosis is any change to the normal microbiome that could change the relationship between the host and associated microbes [@neto_upper_2017]. 
This is well documented gastrointestinal (GI) disorders including irritable bowel syndrome, peptic ulcers, and even gastric and colon cancers [@neto_upper_2017]. 
Gut dysbiosis is documented in patients with colorectal cancers [@nakatsu_alterations_2018]. 

At present, the majority of cancer microbiome studies focus on the bacteria(citation needed). 
Colorectal cancer is previously linked to changes to the colonic bacterial composition yet the gut virome is vastly unexplored [@nakatsu_alterations_2018].

Virome is a viral community associated with a particular ecosystem or holobiont. 
In mammals, the viruses infect host cells and the variety of organisms that inhabit us. 
Identifying and understanding the virome within a host requires genetic and transcriptional identification of the mammal.  This virome may be able to shed light on the host’s genetics in states of health and states of disease. 
The virome is composed of both nucleic acids (DNA and RNA).

Human viruses associated with many cancers, due to their manipulative and mutagenic abilities. (Hannigan, G.D. et al. 2018). 
Viral metagenomic taxa can distinguish colorectal cancer patients from control subjects [@nakatsu_alterations_2018]. 
Furthermore, a subset of markers were identified as high-risk from a subset of patients with colorectal cancer.  

Here we address the knowledge gap with respect to whether bacterial and viral community composition differ between healthy patients and those diagnosed with adenoma and carcinoma. 

We used the data from __hannigan paper___ and __sited in report_ to try to understand the viruses and bacteria in the gut microbiome in colorectal cancer.


# Methods

## Study design and data collection

\todo{study design / data collection based on hannigan and zackular papers}
[@hannigan_diagnostic_2018; @Zackular_human_2014]

The data are available in the NCBI Sequence Read Archive as [PRJNA389927](https://www.ncbi.nlm.nih.gov/bioproject/389927). 
All code used in preparation of this report is available in the following GitHub repository: https://github.com/kelly-sovacool/bioinf545-group3-project.
\todo{make the repo public before submitting}

## 16S rRNA gene sequence processing

## Metagenome and virome quality control

Trimmomatic (v.0.39) [@bolger_trimmomatic_2014] was used to remove adapter sequences and low-quality reads in metagenome and virome sample sets. 
The read quality was assessed using FastQC (v.0.11.9) [@andrews_fastqc_2010] before and after adapter trimming to confirm removal of adapters and low quality reads. 
Unpaired reads were dropped using the repair function in BBTools (v.37.62) [@bushnell_bbtools_2018]. 
Then, reads mapping to the human GRCh38 reference genome were removed using the BWA-MEM algorithm (BWA v.0.7.12) [@li_aligning_2013].
The paired unmapped reads were used for taxonomic profiling and gene annotation.

## Virome assembly

## Classification modeling

## Metagenome taxonomic profiling and gene annotation

The human genome-free reads were used to profile the metagenomic taxonomy at the species, genus, family, and phylum levels using MetaPhlAn2 [@truong_metaphlan2_2015]. 
The results were visualized using the R package phyloseq [@mcmurdie_phyloseq_2013].

The paired reads that did not map to the human genome were aligned to the integrated gene catalog (IGC) [@li_integrated_2014] of 1267 gut microbiome samples consisting of approximately 10 million genes with the BWA MEM algorithm for metagenome annotation [@li_aligning_2013]. 
Annotated genes were extracted from the alignment results using functions geneList and countKegg modified from the MGS-Fast pipeline [@brown_mgs-fast_2019]. 
The differences in gene abundance between healthy, adenoma, and cancer groups were assessed using the R package edgeR [@robinson_edger_2009]. 
The top 500 KEGG numbers of genes that were significantly different between healthy and other groups were selected and the relevant KEGG pathway was determined using the KEGG mapper tool [@kanehisa_kegg_2020].

# Results

\begin{wrapfigure}{R}{0.4\textwidth}
    \begin{center}
        \includegraphics[width=0.35\textwidth]{../figures/rulegraph.png}  
        \caption{Example}
        \label{fig:example}  
    \end{center}
\end{wrapfigure}

Example of referring to a figure in text (Fig. \ref{fig:example}) 
and including it inline with the text.  
\todo{remove example figure before submitting}

# Discussion

# Acknowledgements

Author order was determined by alphabetizing by last name. 
Christina Kang-Yun constructed the workflow for preprocessing metagenome data, taxonomic profiling, and gene annotation and prepared the methods and results for these analyses.
\todo{insert contribution statement here}

# References
