---
title: "Investigating the microbial community of the human gut in colorectal cancer"
author: "Nicole Bowers, Brittany Hicks, Christina Kang-Yun, Katelyn Polemi, Kelly Sovacool"
csl: mbio.csl
output:
  pdf_document:
    includes:
      in_header: preamble_report.tex
    keep_tex: no
    latex_engine: xelatex
    toc: yes
  word_document:
    toc: yes
  html_document:
    df_print: paged
    toc: yes
geometry: margin=1in
link-citations: yes
fontsize: 11pt
bibliography: refs_report.bib
subparagraph: true # fixed error that says "! Argument of \paragraph has an extra }."
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      fig.path = here::here("figures/")
                      )
```
```{r load}
library(broom)
library(here)
library(kableExtra)
library(knitr)
library(tidyverse)
```

\newpage
\todo{Guidelines: https://drive.google.com/file/d/1YNqYiMJYiVWXiSR9yWUbQTPAoUpVa9Id/view}

# Abstract

Colorectal cancer is the third leading cause of cancer related deaths globally [@rawla_epidemiology_2019]. 
Colorectal cancer (CRC) initiates in the large intestine emerging from glandular epithelial cells. 
Due to a selective advantage, obtained from a series of genetic or epigenetic mutations, these unregulated cells grow and can potentially develop into colorectal adenocarcinoma [@ewing_molecular_2014]. 
Many studies have used the cancer microbiome to investigate how these mutations occur, however, they have almost exclusively focused on bacteria. 
Human viruses have been implicated in the development of many cancers such as HPV, HTLV-1 and HIV [@liao_viruses_2006]. 
Through mutagenic and manipulative abilities, viruses cause unregulated growth by disrupting the normal function of cells. 
The association of colorectal cancer and the gut virome remains unknown, therefore, in this study we propose to investigate the differences in the bacterial and viral community composition and their effect on CRC development. 
We performed operational taxonomic unit (OTU) clustering of 16S sequences for bacterial taxonomy and operational viromic unit (OVU) clustering of metaviromes for viral taxonomy to investigate the differences in human colorectal bacterial and viral community composition. 
Using machine learning classification of OTUs and OVUs from healthy or cancerous samples, we aim to compare performance of the random forest model vs logistic regression. 
Overall, the goal of the current study is to classify whether samples are healthy or cancerous based on bacterial and viral taxa. These results provide… 

# Introduction

Colorectal cancer is a leading cause of cancer-related death in the United States and worldwide. Its risk and severity have been linked to colonic bacterial community composition
The leading cancer-related death in the United States is colorectal cancer [@hannigan_diagnostic_2018]. 
Due to screening techniques and improvements in treatment the rate of colorectal cancer has fallen. 
The primary screening technique is colonoscopy. A new technique gaining popularity is Exact Sciences Cologuard, a combination test which looks for both the genetic material found in colon cancer and some more advanced colon polyps and it detects hemoglobin in the stool using fecal immunochemical testing (Cologuard). 
While colon cancer screening techniques are advancing, we aim to understand the environment of colorectal cancer.

Gut dysbiosis is any change to the normal microbiome that could change the relationship between the host and associated microbes [@neto_upper_2017]. 
This is well documented gastrointestinal (GI) disorders including irritable bowel syndrome, peptic ulcers, and even gastric and colon cancers [@neto_upper_2017]. 
Gut dysbiosis is documented in patients with colorectal cancers [@nakatsu_alterations_2018]. 

At present, the majority of cancer microbiome studies focus on the bacteria(citation needed). 
Colorectal cancer is previously linked to changes to the colonic bacterial composition yet the gut virome is vastly unexplored [@nakatsu_alterations_2018].

Virome is a viral community associated with a particular ecosystem or holobiont. 
In mammals, the viruses infect host cells and the variety of organisms that inhabit us. 
Identifying and understanding the virome within a host requires genetic and transcriptional identification of the mammal.  This virome may be able to shed light on the host’s genetics in states of health and states of disease. 
The virome is composed of both nucleic acids (DNA and RNA).

Human viruses associated with many cancers, due to their manipulative and mutagenic abilities [@hannigan_diagnostic_2018]. 
Viral metagenomic taxa can distinguish colorectal cancer patients from control subjects [@nakatsu_alterations_2018]. 
Furthermore, a subset of markers were identified as high-risk from a subset of patients with colorectal cancer.  

Here we address the knowledge gap with respect to whether bacterial and viral community composition differ between healthy patients and those diagnosed with adenoma and carcinoma. 

We used the data from __hannigan paper___ and __sited in report_ to try to understand the viruses and bacteria in the gut microbiome in colorectal cancer.


# Methods

## Study design and data collection

\todo{study design / data collection based on hannigan and zackular papers}
[@hannigan_diagnostic_2018; @zackular_human_2014]

The data are available in the NCBI Sequence Read Archive as [PRJNA389927](https://www.ncbi.nlm.nih.gov/bioproject/389927).
All code used in preparation of this report is available in the following GitHub repository: https://github.com/kelly-sovacool/bioinf545-group3-project.

## 16S rRNA gene sequence processing
The 16S rRNA gene sequences associated with this study were previously reported by [@Zackular_human_2014]Sequence (fastq) and metadata files were downloaded from the following site: http://www.mothur.org/MicrobiomeBiomarkerCRC.
The 16S rRNA gene sequences were analyzed with the mothur software package (v1.37.0) [@kozich_Development_2013] [@schloss_introducing_2009]. Mothur allows contigs from forward and reverse fastq files to be generated, pre-processing steps for primer removal and quality screening, and the identification of unique sequences. After these steps were taken to reduce sequencing and PCR errors, sequences were aligned to the SILVA database [@pruesse_SILVA_2007], screened for chimeras using VSEARCH, and binned into operational taxonomic units (OTUs) using a 97% similarity threshold. 


\todo{mothur}
Alpha and beta diversity metrics were calculated with the vegan R package [@oksanen_vegan_2018].

## Metagenome and virome quality control

Trimmomatic (v.0.39) [@bolger_trimmomatic_2014] was used to remove adapter sequences and low-quality reads in metagenome and virome sample sets. 
The read quality was assessed using FastQC (v.0.11.9) [@andrews_fastqc_2010] before and after adapter trimming to confirm removal of adapters and low quality reads.
Unpaired reads were dropped using the repair function in BBTools (v.37.62) [@bushnell_bbtools_2018].
Then, reads mapping to the human GRCh38 reference genome were removed using the BWA-MEM algorithm (BWA v.0.7.12) [@li_aligning_2013].
The paired unmapped reads were used for taxonomic profiling and gene annotation.

## Virome assembly
Closely related viral contig sequences were categorized into operational viral units (OVUs) in the absence of taxonomic identity. OVUs were defined with the CONCOCT algorithm (v0.4.0). This algorithm applies a variation-based Bayesian approach that bins related contigs by similar tetramer and coabundance profiles within samples [@alneberg_binning_2014].


\todo{virome pipeline}

## Classification modeling

The caret R package [@kuhn_caret_2013] was used to train random forest models to classify samples as healthy or cancerous based on taxonomic abundance data.
Only binary classifier models were trained, rather than three-class models, to reduce the time and computing resources required.
Two models were trained; one with OTU (bacterial) abundances as features and the other with OVU (viral) abundances as features.
For both models, the data were split into training and test sets with 65\% of the data for training.
The models were validated with five-fold cross validation and tuned for mtry values to maximize the area under the receiver operating curve (AUC).

## Metagenome taxonomic profiling and gene annotation

The human genome-free reads were used to profile the metagenomic taxonomy at the species, genus, family, and phylum levels using MetaPhlAn2 [@truong_metaphlan2_2015].
The results were visualized using the R package phyloseq [@mcmurdie_phyloseq_2013].

The paired reads that did not map to the human genome were aligned to the Integrated Gene Catalog (IGC) [@li_integrated_2014] of 1,267 gut microbiome samples consisting of approximately 10 million genes with the BWA MEM algorithm for metagenome annotation [@li_aligning_2013]. 
Annotated genes were extracted from the alignment results using functions geneList and countKegg modified from the MGS-Fast pipeline [@brown_mgs-fast_2019]. 
The differences in gene abundance between healthy, adenoma, and cancer groups were assessed using the R package edgeR [@robinson_edger_2009]. 
Up to 480 top KEGG numbers of genes that were significantly different between healthy and other groups were selected and the relevant KEGG pathway was determined using the KEGG mapper tool [@kanehisa_kegg_2020].

# Results

## Bacterial diversity

\begin{figure}[ht]
  \includegraphics[width=1\textwidth, center]{../figures/alpha_beta_diversity.png}  
  \caption{(A) Shannon diversity index across disease states and (B) Multi-dimensional scaling plot of Bray-Curtis Distances.}
  \label{fig:diversity}
\end{figure}

The influence of colorectal cancer on bacterial OTU diversity was evaluated amongst the three disease classes. A total of 3,904 OTUs were identified. 
Shannon diversity index of the 16S bacterial abundance data showed a general decreasing trend along the progression from healthy, to adenoma, to cancer (Fig. \ref{fig:diversity}A). 
As a metric of alpha diversity, the Shannon diversity index describes diversity within communities in terms of richness and evenness.
In contrast to Shannon diversity, Bray-Curtis distance is a beta diversity metric that quantifies diversity between communities [@tuomisto_diversity_2010].
The pairwise Bray-Curtis distances of sample-wise OTU abundances were calculated and plotted as a multi-dimensional scaling plot, but there is no distinguishable separation among the disease classes (Fig. \ref{fig:diversity}B).
This is further supported by analysis of variance (ANOVA) analyses for both alpha and beta diversity (P value > 0.23 and P value > 0.3 respectively).
These alpha and beta diversity metrics characterize and distinguish communities in broad strokes.

## Classification modeling

\begin{wrapfigure}{r}{0.4\textwidth}
    \begin{center}
        \includegraphics[width=0.39\textwidth]{../figures/auroc.png}  
        \caption{Model performance as measured by the receiver operating characteristic (ROC) curve for models trained on bacterial abundance or viral abundance data.}
        \label{fig:auroc}  
    \end{center}
\end{wrapfigure}

Random forest models were trained on OTU abundances from 16S bacterial sequence data and on OVU abundances from viral metagenomes to classify samples as healthy or cancerous. 
Both models had modest performance with an AUROC of 0.765 for the bacterial model and an AUROC of 0.73 for the viral model (Fig. \ref{fig:auroc}). 
The bacterial abundance data consisted of 3,04 OTUs, while the viral abundance data contained only 70 OVUs.
This discrepancy in the number of features as inputs to the random forest models may have contributed to the bacterial model outperforming the viral model for classifying samples as healthy or cancerous.
An OTU containing members of the genus _Fusobacterium_ was by far the most important feature of the bacterial model.
The viral metagenomes were not aligned to a reference database, so OVUs were not able to be identified for feature importance determination.
Previous work by Hannigan _et al._ additionally included models trained on bacterial whole metagenomes and models combining bacterial and viral abundance data.

## Taxonomonic abundance

\todo{moar werdz go here}

\begin{figure}[H]
  \includegraphics[width=1\textwidth, center]{../figures/metag_taxa_heatmap_phy_fam.png}  
  \caption{}
  \label{fig:taxabun}
\end{figure}

## Metagenome annotation and quantification

\begin{wrapfigure}{l}{0.5\textwidth}
    \begin{center}
        \includegraphics[width=0.49\textwidth]{../figures/gene_abundance_MDS.pdf}  
        \caption{Multidimensional scaling (MDS) abundance plot based on KEGG annotation counts derived from gut microbiome of patients that are healthy (H) or diagnosed with adenoma (A) or cancer (C).}
        \label{fig:MDS_KeggNo}  
    \end{center}
\end{wrapfigure}

The gut metagenome from patients that are healthy or diagnosed with adenoma or cancer was aligned to the IGC database to annotate the genes with KEGG numbers.
KEGG number counts were calculated for all samples, then were used to determine whether specific genes were more or less abundant in patients with adenoma or cancer as compared to those who are healthy using edgeR.
Seven samples, including the negative control, with extremely low read counts were excluded from subsequent analyses, as they strongly skewed the results.
Multidimensional scaling (MDS) analysis of the KEGG number counts between different groups did not show clear clustering within groups (Fig. \ref{fig:MDS_KeggNo}).

In the subsequent differential gene abundance analysis, abundance of 6,104 distinct genes were compared between healthy vs adenoma and healthy vs cancer patients. 
In both cases, there were approximately 130 genes that were less abundant in adenoma and cancer patients, whereas around 600 to 1,000 genes were more abundant. 
Differentially abundant genes that were found in both adenoma and cancer patient groups were selected and used to map to metabolic pathways using the KEGG mapper (see Tables \ref{tab:genes_neg} and \ref{tab:genes_pos}). 

The less and more abundant genes found in adenoma and cancer patients mapped to approximately 20 and 550 genes in the KEGG pathways, respectively. 
Less abundant genes in adenoma and cancer patients were slightly enriched in pathways in metabolism, genetic information processing, signal transduction, transport and catabolism, immune system, and bacterial infectious disease (Table \ref{tab:genes_neg}).
Contrastingly, more abundant genes in adenoma and cancer patients were most enriched in pathways in metabolism, environmental information processing, and cellular proccesses amongst other pathways. 
Though human genome was filtered out through pre-processing, pathways pertaining to human diseases and organismal systems were also enriched in the case of genes more abundant in adenoma and cancer patients.


```{r kegg_genes_neg}
DEgenes_neg <-
  read.csv(file = here(
    "data",
    "metagenome",
    "gene_abundance_results",
    "DEgenes_neg.csv"
  ))
DEgenes_neg$Number.of.Genes <-
  DEgenes_neg$Number.of.Genes %>% replace_na("")
kable(
  DEgenes_neg,
  col.names = c("KEGG Pathway", "", "Number of Genes", "KEGG No."),
  caption = "Genes that are less abundant in adenoma or cancer patients as compared to healthy patients. \\label{tab:genes_neg}"
) %>%
  kable_styling(font_size = 8) %>%
  row_spec(c(1, 17, 20, 22, 24, 26), bold = TRUE)
```


```{r kegg_genes_pos}
DEgenes_pos <-
  read.csv(file = here(
    "data",
    "metagenome",
    "gene_abundance_results",
    "DEgenes_pos.csv"
  ))
DEgenes_pos <- DEgenes_pos %>% replace(., is.na(.), '')
DEgenes_pos <-
  DEgenes_pos %>% filter(Total.Number.of.Genes != "" | KEGG.Pathway != "")
DEgenes_pos$etc <- as.numeric(DEgenes_pos$Total.Number.of.Genes) - as.numeric(DEgenes_pos$Number.of.Genes)
DEgenes_pos$X <- paste(DEgenes_pos$X, if_else(DEgenes_pos$etc == 0, "", ", etc.", ""), sep = '')
DEgenes_pos <- DEgenes_pos %>% select(KEGG.Pathway, X, Total.Number.of.Genes)
kable(
  DEgenes_pos,
  col.names = c("KEGG Pathway", "", "Number of Genes"),
  caption = "Genes that are more abundant in adenoma or cancer patients as compared to healthy patients. \\label{tab:genes_pos}"
) %>%
  kable_styling(font_size = 8) %>%
  row_spec(c(1, 14, 18, 21, 25, 28), bold = TRUE)
```


# Discussion

Based upon the alpha and beta diversity of the OTUs, it is possible that the gut microbiome does not significantly shift in response to the presence of colorectal cancer. 
Standard alpha and beta diversity metrics do not adequately capture community differences of the bacterial community. 
This illuminates that examining the 16S sequences of the gut microbiome may not be an adequate tool for identifying a healthy versus a cancerous colon.

\todo{discuss: taxonomic abundance, random forest discriminates disease classes better than diversity metrics, gene abundance}

# Acknowledgements

Author order was determined by alphabetizing by last name.
Christina Kang-Yun constructed the workflow for preprocessing metagenome data, taxonomic profiling, and gene annotation and prepared the methods and results for these analyses.
\todo{insert contribution statement here}

# References
